File: 1616-minimum-difference-between-largest-and-smallest-value-in-three-moves/minimum-difference-between-largest-and-smallest-value-in-three-moves.java
import java.util.Arrays;

class Solution {
    public int minDifference(int[] nums) {
        int n = nums.length;
        if (n <= 4) {
            return 0;
        }

        Arrays.sort(nums);

        int x = Integer.MAX_VALUE;
        for (int i = 0; i < 4; i++) {
            x = Math.min(x, nums[n - 4 + i] - nums[i]);
        }

        return x;
    }
}

File: 1616-minimum-difference-between-largest-and-smallest-value-in-three-moves/minimum-difference-between-largest-and-smallest-value-in-three-moves.py
class Solution:
    def minDifference(self, nums):
        n = len(nums)
        if n <= 4:
            return 0

        nums.sort()

        x = float('inf')
        for i in range(4):
            x = min(x, nums[-4 + i] - nums[i])

        return x
